### PR TITLE
Offline mode

### PR DESCRIPTION
**Pull request by ramya-rao-a**
This PR implements features detailed in #54354 

Adding below folks as reviewers
@sbatten for Menu entries
@joaomoreno for vscode updates
@sandy081 for extension view and updates
@JacksonKearl for extension recommendations
@mjbvz for Release notes and logUploader

Pending items to be taken up in another PR
- Disabling bing powered settings search in offline mode
- Disabling the badges denoting existence of updates
- Fix the issue with the update entries in the menus
- Ensure built in extensions comply with offline mode

### CODE REVIEW

**Comment by ramya-rao-a** [comment]
Adding below folks as reviewers
@sbatten for Menu entries
@joaomoreno for vscode updates
@sandy081 for extension view and updates
@JacksonKearl for extension recommendations

**Comment by sbatten** [comment]
menu stuff LGTM

File: src/vs/platform/actions/common/offlineMode.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		});
+	}
+
+	private updateEnabled() {
+		this.enabled = this.configurationService.getValue(offlineModeSetting) !== true;
+	}
+
+	run(): TPromise<any> {
+		if (this.storageService.getBoolean(this.disclaimerStorageKey, StorageScope.GLOBAL, false) === false) {
+			this.notificationService.prompt(Severity.Info, localize('offlineModeDisclaimer', "VS Code cannot stop extensions from making network requests in offline mode. If extensions make such requests, please log an issue against them."), [
```
Comment (mjbvz) [inline src/vs/platform/actions/common/offlineMode.ts]: Is there a way for extensions to check if the user in offline mode?

**Review by mjbvz — APPROVED** [review]
Release notes changes look good. Thanks!

File: src/vs/platform/actions/common/offlineMode.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/vs/platform/actions/common/offlineMode.ts]: @mjbvz Yes, they can look at the value for the `workbench.enableOfflineMode` setting. I havent yet gone through built-in extensions to ensure they are able to comply with the offline mode. That will be another PR.

Comment (mjbvz) [reply src/vs/platform/actions/common/offlineMode.ts]: Ok, so that will added to `vscode.d.ts`?

**Comment by JacksonKearl** [comment]
Extensions looks good.

File: src/vs/workbench/electron-browser/main.contribution.ts [hunk 2]
```diff
@@ -489,6 +489,12 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('settingsTocVisible', "Controls whether the settings editor Table of Contents is visible."),
 			'default': true,
 			'scope': ConfigurationScope.WINDOW
+		},
+		'workbench.enableOfflineMode': {
+			'type': 'boolean',
+			'description': nls.localize('settingsOfflineMode', "Enables offline mode where no requests are made over the network by VS Code."),
```
Comment (JacksonKearl) [inline src/vs/workbench/electron-browser/main.contribution.ts]: I wonder if this wording should be changed to say that no requests are made *to Microsoft*, as we do still make requests to SCM providers, and there's nothing we can do to stop extensions (so far as I know)

File: src/vs/platform/actions/common/offlineMode.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/vs/platform/actions/common/offlineMode.ts]: Read the setting value directly

**Pull request closed without merging**